test/view_component/view_component_test.rb

```diff
@@ -386,6 +386,84 @@ def test_backtrace_returns_correct_file_and_line_number
     assert_match %r[app/components/exception_in_template_component\.html\.erb:2], error.backtrace[0]
   end
 
+  # Some example usages:
+  #
+  # <%= render(ProductComponent.all(@products)) %>
+  # <%= render(ProductComponent.all(collection: @products, as: :product, foo: foo)) %>
+  # <%= render(ProductComponent.new(product: @product)) %>
+
+  def test_render_collection
+    @products = [OpenStruct.new(title: "Hi"), OpenStruct.new(title: "Bye")]
+    render_inline(ProductComponent.all(collection: @products, extra: "extra"))
+
+    assert_selector("h1", count: 2)
+  end
+
+  def test_render_collection_minimal
```

Comment:
Can we add assertions that prove the extra arguments get passed to the components? At the moment this test would pass if `extra: "extra"` wasn't passed in.

Reply:
This one went away.